src/gurt/hash.c

```diff
@@ -475,8 +475,10 @@ d_hash_rec_insert(struct d_hash_table *htable, const void *key,
 
 	if (exclusive) {
 		tmp = ch_rec_find(htable, bucket, key, ksize, D_HASH_LRU_NONE);
-		if (tmp)
+		if (tmp) {
+			D_ERROR("Dup key detected\n");
```

Comment:
A D_DEBUG would be better. For many cases, -DER_EXIST and -DER_NONEXIST are not real errors, depending on callers' protocols.